File: src/main/java/com/rapidprototyping/manbot/controller/NewUserListener.java
package com.rapidprototyping.manbot.controller;

import net.dv8tion.jda.core.entities.User;
import net.dv8tion.jda.core.events.guild.member.GuildMemberJoinEvent;
import net.dv8tion.jda.core.hooks.ListenerAdapter;

/**
 * @author Pavan C (pavan407)
 */
public class NewUserListener extends ListenerAdapter
{
    @Override
    public void onGuildMemberJoin(GuildMemberJoinEvent event)
    {
        User newUser = event.getUser();
        event.getGuild().getTextChannelById(403187432159313923L).sendMessage("Please welcome " +
                newUser.getAsMention() + " to the guild!").queue();
    }
}

File: bot/src/main/kotlin/com/manbot/NewUserListener.kt
package com.manbot

import net.dv8tion.jda.core.events.guild.member.GuildMemberJoinEvent
import net.dv8tion.jda.core.hooks.ListenerAdapter

/**
 * @author Pavan C. (pavan407)
 */
class NewUserListener : ListenerAdapter()
{
    override fun onGuildMemberJoin(event: GuildMemberJoinEvent)
    {
        event.guild.getTextChannelById(403187432159313923L).sendMessage("Please welcome " +
                event.user.asMention + " to the guild!").queue()
    }
}
